src/Microsoft.HttpRepl/Commands/BaseHttpCommand.cs

```diff
@@ -457,7 +463,9 @@ private static async Task FormatBodyAsync(DefaultCommandInput<ICoreParseResult>
                     {
                         try
                         {
+#pragma warning disable CA2012 // Use ValueTasks correctly
```

Comment:
Oof, why is this complaining?  ValueTasks are fragile creatures if not used correctly.  You must consume their result once, and there are possible perf implications (or bugs) if you don't.  See https://github.com/dotnet/roslyn-analyzers/issues/3617#issuecomment-625986124 and https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/#valid-consumption-patterns-for-valuetasks.

Reply:
Yeah, it seemed to be complaining for a valid reason. I'm just not sure yet exactly how I can back out its usage. `WaitForCompletionAsync` checks `readTask` to see if it is completed in a loop (which seems bad, based on the documentation) while also checking to see if there are any keystrokes available in the console and bailing when either thing happens. 
I _think_ I can just drop that altogether and let the cancellation token be the only thing that cancels the read (the token should get canceled if you ctrl+c during the operation). But I wanted to do some testing on that specifically and didn't think that should block the rest of the PR. That's why I added #418 to make sure I came back to it.

Reply:
Actually, looking at the second link with the scenarios you shouldn't do, I'm not sure why it's complaining. I thought it was the multiple checks on IsCompleted and the multiple checks on Result. But those aren't called out. It does seem to fail the short rule though:
> The short rule is this: with a ValueTask or a ValueTask<TResult>, you should either await it directly (optionally with .ConfigureAwait(false)) or call AsTask() on it directly, and then never use it again

Regardless, I think I'll be able to remove that extra work and just await it. Just need to confirm some things separately.

Reply:
I'd be willing to bet that the problem is that you're passing it off to another method - it is a struct after all, so you're not consuming the ValueTask in this scope, you're consuming the copy-by-value'd clone of it in the other method.  (At least, you don't consume it in all code paths here... the `.Result` call is inside an if).

If you're going to pass it around, you should probably `.AsTask()` it first, and then it would behave as expected.

(Really, the fact that you're passing a ValueTask around at all seems like a bug that violates the implicit ValueTask contract.)

Reply:
I went ahead and removed the suppression and used `.AsTask()` to make it safer while I figure out if I can remove the extra waits and checks for keystrokes. Updated #418 a little bit to note what I'm doing.